Sources/SKTestSupport/TestSourceKitLSPClient.swift

```diff
@@ -216,7 +217,7 @@ package final class TestSourceKitLSPClient: MessageHandler, Sendable {
     // It's really unfortunate that there are no async deinits. If we had async
     // deinits, we could await the sending of a ShutdownRequest.
     let sema = DispatchSemaphore(value: 0)
-    server.handle(ShutdownRequest(), id: .number(Int(nextRequestID.fetchAndIncrement()))) { result in
+    server.handle(ShutdownRequest(), id: .number(Int(nextRequestID.add(1, ordering: .sequentiallyConsistent).newValue))) { result in
```

Comment:
```suggestion
    server.handle(ShutdownRequest(), id: .number(Int(nextRequestID.add(1, ordering: .sequentiallyConsistent).oldValue))) { result in
```

Reply:
Yes, that’s correct. It’s just this case where we want to use the `oldValue` because `nextRequestID` is the next request ID that hasn’t been used yet. We use `nextRequestID` with `oldValue` in other places as well.